tools/autotag/tag_script.py

```diff
@@ -0,0 +1,272 @@
+        type=str,
+        help=(
+            "The access token to use for PRs, if different from the other"
+            " token"
+        ),
+    )
+    parser.add_argument(
+        "-l", "--library", help="A library to release.", action="append"
+    )
+    parser.add_argument(
+        "-x",
+        "--exclude",
+        dest="_exclude",
+        help="A library to exclude from release.",
+        action="append",
+    )
+    parser.add_argument(
+        "-o",
+        "--org",
+        help="The Github org to look for the repos in.",
+    )
+    add_arg_pair(parser, "release", "the tag & release.")
+    add_arg_pair(parser, "pulls", "the pull requests to internal repos.")
+    add_arg_pair(parser, "previous", " use previous versions as required.")
+    parser.add_argument(
+        "--manifest-url",
+        help="The url to download the manifest.xml file from.",
+    )
+    parser.add_argument(
+        "--branch", help="A branch to check the changelog from."
+    )
+    parser.add_argument(
+        "--github-url",
+        help="The GitHub URL to use for tag & release and pull requests.",
+        default="github.com",
+    )
+    parser.add_argument(
+        "--compile_file",
+        help="The file to write the compiled release notes to.",
+        type=argparse.FileType("w", encoding="utf-8"),
+        default=None,
+    )
+    parser.add_argument(
+        "version", help="The ROCm release version to release libraries for."
+    )
+    args = parser.parse_args(namespace=TaggingArgs())
+    if args.pr_token is None:
+        args.pr_token = args.token
+    return args
+
+
+def run_tagging():
+    """Run the tagging/releasing process on each specified library."""
+    args = parse_args()
+
+    if args.manifest_url is None:
+        manifest_url = (
+            "https://github.com/RadeonOpenCompute/ROCm/raw/develop/default.xml"
+        )
+    else:
+        manifest_url = args.manifest_url
+    manifest_path, _ = urllib.request.urlretrieve(manifest_url, "manifest.xml")
+    manifest_tree = ET.parse(manifest_path).getroot()
+
+    gh_args = {
+        "login_or_token": get_token(
+            args.token, token_keys=["gh_token", "token"]
+        )
+    }
+    pr_args = {"login_or_token": get_token(args.token, token_keys="pr_token")}
+
+    if args.github_url != "github.com":
+        gh_args["base_url"] = pr_args[
+            "base_url"
+        ] = f"https://{args.github_url}/api/v3"
+
+    data_factory = ReleaseDataFactory(
+        args.org, args.version, Github(**gh_args), Github(**pr_args)
+    )
+
+    remote_map = {
+        "rocmswplatform": "ROCmSoftwarePlatform",
+        "rocmswplatform-internal": "ROCmSoftwarePlatform",
+        "amdgpuopenlib": "GPUOpen-ProfessionalCompute-Libraries",
+        "radeonopencompute": "RadeonOpenCompute",
+        "radeonopencompute-internal": "RadeonOpenCompute"
+    }
+
+    libraries: list[ReleaseLib] = []
+
+    for entry in manifest_tree.findall(".*[@groups='mathlibs']"):
```

Comment:
I'm rebasing my PR onto this to merge the scripts. This won't work with how the manifest `default.xml` is setup in this reposotiry. Is it OK to add this property?